catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -83,28 +81,4 @@ public static boolean logException(Throwable t) {
 
 		return false;
 	}
-
-	public static void sendUnhandledCaughtException() {
-		String json = preferences.getString(EXCEPTION_FOR_REPORT, "");
-		if (isReportingEnabled() && !json.isEmpty()) {
-			Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
-			Gson gson = new Gson();
-			Throwable exception = gson.fromJson(json, Throwable.class);
-			logException(exception);
-			preferences.edit().remove(EXCEPTION_FOR_REPORT).commit();
-		}
-	}
-
-	public static void storeUnhandledException(Throwable exception) {
-		SharedPreferences.Editor prefsEditor = preferences.edit();
-		if (isReportingEnabled()) {
-			Gson gson = new Gson();
-			String check = preferences.getString(EXCEPTION_FOR_REPORT, "");
-			if (check.isEmpty()) {
-				String json = gson.toJson(exception);
-				prefsEditor.putString(EXCEPTION_FOR_REPORT, json);
-				prefsEditor.commit();
-			}
-		}
-	}
```

Comment:
We're using a custom crash recovering process (flow). The CrashReporter partially handles it by storing and sending the stored exception. IMO the CrashReporter is not responsible for that. So I removed the code only needed for crash recovering here and but it into it's own class.

The CrashReporter just reports (sends) exceptions now. Single Responsibility.